presto-main/src/test/java/io/prestosql/type/TestTimestampLegacy.java

```diff
@@ -62,4 +69,17 @@ public void testCastFromSlice()
                 TIMESTAMP,
                 sqlTimestampOf(2001, 1, 21, 9, 0, 0, 0, session));
     }
+
+    @Test
+    public void testInvalidLiteral()
+    {
+        Session session = testSessionBuilder()
+                .setSystemProperty("legacy_timestamp", "true")
+                .setTimeZoneKey(TimeZoneKey.getTimeZoneKey("America/Bahia_Banderas"))
```

Comment:
Use a zone different from test jvm zone (to cover the fact that jvm zone doesn't impact this).
eg Vilnius
https://github.com/prestosql/presto/blob/ca878d471740c07155ea29fb74fe8be0ba855a68/presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlTypeMapping.java#L129-L130

(i used Bahia in my previous comment, because i remember when some gap is in Bahia, sorry for misguiding you)

Reply:
Does it matter? We rely on other tests to ensure that the session timezone is being respected. This is just making sure that "assuming session timezone is being respected => this timestamp is invalid". If the session timezone is not being respected and the test passes due to VM timezone, then that should trigger failures elsewhere.

Reply:
It does matter. For example timestamp could be detected as invalid if it's invalid in JVM zone, but then could be shifted if it is invalid in the session zone.

 Actually, i hope we also do have a test that parses timestamp being  a gap in JVM zone (but not in the session zone) and verifies it was succesful. We definitely do this in connector tests, but not sure if here.